### PR TITLE
Snoflake 2.1.2 upgrade

### PR DESCRIPTION
**Pull request by EduardHantig**

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/Utils.java [hunk 1]
```diff
@@ -393,12 +435,12 @@ static String validateConfig(Map<String, String> config) {
...
+      if (config.containsKey(SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_MAX_CLIENT_LAG)) {
         invalidConfigParams.put(
-            SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_FILE_VERSION,
+            SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_MAX_CLIENT_LAG,
             Utils.formatString(
                 "{} is only available with ingestion type: {}.",
-                SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_FILE_VERSION,
+                SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_MAX_CLIENT_LAG,
                 IngestionMethodConfig.SNOWPIPE_STREAMING.toString()));
       }
```
Comment (JonathonO) [inline src/main/java/com/snowflake/kafka/connector/Utils.java:445]: @EduardHantig are you sure the `SNOWPIPE_STREAMING_FILE_VERSION` is replaced by `SNOWPIPE_STREAMING_MAX_CLIENT_LAG`?

**Review by JonathonO — COMMENTED** [review]
@EduardHantig can you prepare a build for me, please? You will probably need to add this to the `mvn` arguments `-DskipTests -DskipITs -Dgpg.skip=true`

File: src/main/java/com/snowflake/kafka/connector/Utils.java (hunk 1 above)
Comment (EduardHantig) [reply src/main/java/com/snowflake/kafka/connector/Utils.java:445]: @JonathonO I have another PR where all changes are just pulled without solving conflicts. You can check here is the same also - https://github.com/streamkap-com/snowflake-kafka-connector/pull/16

Meanwhile I also checked git annotation of git blame:
Screenshot 2024-02-01 at 14 17 14

**Comment by EduardHantig** [comment]
> @EduardHantig can you prepare a build for me, please? You will probably need to add this to the `mvn` arguments `-DskipTests -DskipITs -Dgpg.skip=true`

@JonathonO In order to make the build succeeding, I commented out the two classes I've discussed about on previous comment (i.e. `SnowflakeStreamkapSinkIT`, `StreamkapITSnowflake`)
I have the jars created.

**Comment by JonathonO** [comment]
> > @EduardHantig can you prepare a build for me, please? You will probably need to add this to the `mvn` arguments `-DskipTests -DskipITs -Dgpg.skip=true`
> 
> @JonathonO In order to make the build succeeding, I commented out the two classes I've discussed about on previous comment (i.e. `SnowflakeStreamkapSinkIT`, `StreamkapITSnowflake`) I have the jars created.

You shouldn't have to comment those out if you skip ITs in the `mvn` build, surely? That's how I've done it in the past anyway. @acristu is there some reason why your ITs wouldn't be skipped by `DskipITs`? 

We can't be commenting out classes to do builds

**Comment by acristu** [comment]
> SnowflakeStreamkapSinkIT

@JonathonO / @EduardHantig  this test should not be skipped, it is the only one that actually runs on our github actions: https://github.com/streamkap-com/streamkap-snowflake-connect-pkg/actions/runs/6985004394/job/19008688898

**Comment by JonathonO** [comment]
Superseded by #18

**Pull request closed without merging**